packages/cli/src/rollup-plugins/typescript-declarations/common.ts

```diff
@@ -113,8 +113,17 @@ let memoizedGetProgram = weakMemoize((typescript: TS) =>
   })
 );
 
-export async function getProgram(dirname: string, pkgName: string, ts: TS) {
-  let configFileName = ts.findConfigFile(dirname, ts.sys.fileExists);
+export async function getProgram(
+  dirname: string,
+  pkgName: string,
+  ts: TS,
+  configName?: string
+) {
+  let configFileName = ts.findConfigFile(
```

Comment:
Could you change this to use `path.resolve(dirname, tsconfigPath)` instead of `ts.findConfigFile` when a tsconfig path is provided? (probably call `ts.sys.fileExists` on the resulting path and throw an error if the file doesn't exist)

Reply:
And add a test involving e.g. a tsconfig in some directory in the package

Reply:
Hey @emmatown, from my understanding you would like the `preconstruct` configuration `tsconfig` entry to be a `path` a not a `filename` ?

Unless I'm mistaken, `ts.findConfigFile` will already recursively search in parents folder for the `configName` named `tsconfig` and check if it exists.
In my opinion this might be interesting to keep this logic governance in this method and only configure the `tsconfig` as a filename ?

Please let me know

Reply:
> And add a test involving e.g. a tsconfig in some directory in the package

Unless I'm mistaken such a test already exists ?
https://github.com/preconstruct/preconstruct/pull/618/files#diff-7ce7cff34f111fae374f2e14cec1f61113b7f2d3427085d6a060254ce449a495R1424

Reply:
I've implemented in the latest commits what you suggested assuming you wanted to enable users to prefix the `tsconfig` filename with a custom `path`
```
{
  preconstruct: {
    tsconfig: 'some-folder/tsconfig.build.json'
  }
}
```